src/components/structures/MatrixChat.js

```diff
@@ -592,23 +575,40 @@ module.exports = React.createClass({
         this.scrollStateMap[roomId] = state;
     },
 
-    onLoggedIn: function(credentials) {
+    _doLogin(creds) {
+        Lifecycle.login(creds, {
+            syncTimelineLimit: this.props.config.sync_timeline_limit,
+        });
+    },
+
+    _onHaveCredentials: function(credentials) {
         credentials.guest = Boolean(credentials.guest);
-        console.log("onLoggedIn => %s (guest=%s)", credentials.userId, credentials.guest);
-        MatrixClientPeg.replaceUsingAccessToken(
-            credentials.homeserverUrl, credentials.identityServerUrl,
-            credentials.userId, credentials.accessToken, credentials.guest
-        );
+        Lifecycle.login(credentials);
+    },
+
+    _onLoggedIn: function(credentials) {
```

Comment:
a comment saying when this is called (ie, just after a new MatrixClient is constructed with the new access creds) would be useful